tensorboard/plugins/graph/tf_graph_loader/tf-graph-loader.ts

```diff
@@ -0,0 +1,275 @@
+        const graphPath =
+            tf_backend.getRouter().pluginRoute('graphs', '/graph', params);
+        return this._fetchAndConstructHierarchicalGraph(
+            graphPath, null, overridingHierarchyParams).then(() => {
+              this._graphRunTag = {run, tag};
+            });
+      }
+      case tf.graph.SelectionType.PROFILE: {
+        const {tags} = this.datasets.find(({name}) => name === run);
+        const tagMeta = tags.find(t => t.tag === tag);
+        // In case current tag misses opGraph but has profile information,
+        // we fallback to the v1 behavior of fetching the run graph.
+        const requiredOpGraphTag = tagMeta.opGraph ? tag : null;
+
+        console.assert(
+            tags.find(t => t.tag === requiredOpGraphTag),
+            `Required tag (${requiredOpGraphTag}) is missing.`);
+
+        const shouldFetchGraph = !this._graphRunTag ||
+            this._graphRunTag.run !== run ||
+            this._graphRunTag.tag !== requiredOpGraphTag;
+        const maybeFetchGraphPromise = shouldFetchGraph ?
+            this._load({
+              run,
+              tag: requiredOpGraphTag,
+              type: tf.graph.SelectionType.OP_GRAPH,
+            }) : Promise.resolve();
+        const params = new URLSearchParams();
+        params.set('tag', tag);
+        params.set('run', run);
+        const metadataPath = tf_backend.getRouter().pluginRoute(
+            'graphs', '/run_metadata', params);
+        return maybeFetchGraphPromise
+            .then(() => this._readAndParseMetadata(metadataPath));
+      }
+      default:
+        return Promise.reject(new Error(`Unknown selection type: ${selectionType}`));
+    }
+  },
+  _readAndParseMetadata: function(path) {
+    // Reset the progress bar to 0.
+    this.set('progress', {
+      value: 0,
+      msg: ''
+    });
+    var tracker = tf.graph.util.getTracker(this);
+    tf.graph.parser.fetchAndParseMetadata(path, tracker)
+        .then((stats) => {
+          this._setOutStats(stats);
+        });
+  },
+  /**
+   * @param {?string} path
+   * @param {string=} pbTxtFile
+   * @param {Object=} overridingHierarchyParams
+   * @return {!Promise<void, Error>}
+   */
+  _fetchAndConstructHierarchicalGraph: function(
+      path, pbTxtFile, overridingHierarchyParams) {
+    // Reset the progress bar to 0.
+    this.set('progress', {
+      value: 0,
+      msg: ''
+    });
+    var tracker = tf.graph.util.getTracker(this);
+    var hierarchyParams = {
+      verifyTemplate: true,
+      // If a set of numbered op nodes has at least this number of nodes
+      // then group them into a series node.
+      seriesNodeMinSize: 5,
+      // A map of series node names to series grouping settings, to indicate
+      // if a series is to be rendered as grouped or ungrouqped.
+      // Starts out empty which allows the renderer to decide which series
+      // are initially rendered grouped and which aren't.
+      seriesMap: {},
+      rankDirection: 'BT',
```

Comment:
Lines 180–181 are new in this commit. Why?

Reply:
Yeah and shrug. This component was consuming the API incorrectly and it was complaining about wrong type.

Reply:
Got it; thanks!